full-node/db/sov-db/src/native_db.rs

```diff
@@ -31,29 +36,72 @@ impl NativeDB {
             &gen_rocksdb_options(&Default::default(), false),
         )?;
 
+        let next_version = Self::last_version_written(&inner)?.unwrap_or_default() + 1;
         Ok(Self {
             db: Arc::new(inner),
+            next_version: Arc::new(Mutex::new(next_version)),
         })
     }
 
     /// Queries for a value in the [`NativeDB`], given a key.
-    pub fn get_value_option(&self, key: &StateKey) -> anyhow::Result<Option<Vec<u8>>> {
-        self.db
-            .get::<ModuleAccessoryState>(key)
-            .map(Option::flatten)
+    pub fn get_value_option(
+        &self,
+        key: &AccessoryKey,
+        version: Option<Version>,
+    ) -> anyhow::Result<Option<Vec<u8>>> {
+        let version_to_use = version.unwrap_or_else(|| self.get_next_version());
+        let mut iter = self.db.iter::<ModuleAccessoryState>()?;
```

Comment:
With this change, we use iterator on every access to the database. 

Do we want to measure performance impact on regular execution?

Maybe it is too late to discuss it, but it looks like archival state taxes on regular execution.

I am raising this point, because together with ForkManagement our storage can become bottleneck.